prime-dotnet-webapi/Controllers/HealthAuthoritySitesController.cs

```diff
@@ -251,5 +251,30 @@ public async Task<ActionResult> HealthAuthoritySiteSubmission(int healthAuthorit
 
             return NoContent();
         }
+
+        // HEAD: api/health-authorities/5/sites/duplicate-pec?pec=abc
+        /// <summary>
+        /// Checks if the provided PEC is already assigned to another site under another Health Authority
+        /// </summary>
+        /// <param name="healthAuthorityId"></param>
+        /// <param name="pec"></param>
+        [HttpHead("duplicate-pec", Name = nameof(CheckPecExistsInOtherHealthAuthority))]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult> CheckPecExistsInOtherHealthAuthority(int healthAuthorityId, [FromQuery] string pec)
+        {
+            if (string.IsNullOrEmpty(pec))
+            {
+                return NotFound();
+            }
+            if (await _healthAuthoritySiteService.CheckForDuplicatePecAsync(healthAuthorityId, pec))
+            {
+                return Ok();
+            }
+
+            return NotFound();
+        }
```

Comment:
Why make a new endpoint + service method instead of updating the existing code to determine if a PEC is assignable?

Reply:
That could be done. I just tried keep the HA end point in the HA Sites controller. Also, I just looked at the `PecAssignable()` end point and the `PecAssignableAsync()` method. I'm not sure those are easier to edit and use than to just add the new endpoint ands service method. We can chat about that when you have a minute.

Reply:
Now using the existing end point and service method for `pecAssignable`